tests/test_job_tracker_v2.py

```diff
@@ -751,9 +927,65 @@ def test_parse_application_report_running(self):
             "memory": {"unit": "mb-seconds", "value": 96183879},
         }
 
-    def test_parse_application_report_empty(self):
+    def test_parse_application_response_empty(self):
         with pytest.raises(YarnAppReportParseException):
-            _ = YarnStatusGetter().parse_application_report(report="")
+            _ = YarnStatusGetter.parse_application_response(json={})
+
+    def test_get_authentication_provider_use_kerberos_on(self, monkeypatch):
+        """Simple check that the config has the desired effect: we get the expected auth object."""
+        monkeypatch.setenv("YARN_AUTH_USE_KERBEROS", "yes")
+        auth = YarnStatusGetter().get_authentication_provider()
+        assert isinstance(auth, requests_gssapi.HTTPKerberosAuth)
```

Comment:
If you pass auth to the constructor of `YarnStatusGetter` (be it a `bool`/`HTTPKerberosAuth`/`None`) this test is covered by `test_kerberos_auth_should_fail_when_there_is_no_kerberos_setup` below so you can get rid of this test and the `get_authentication_provider` method as well.

Reply:
Indeed. In the end I was able to remove all three of these tests.